src/typeguard/_functions.py

```diff
@@ -260,6 +260,25 @@ def check_yield_type(yieldval: T, memo: CallMemo) -> T:
     return yieldval
 
 
+def check_variable_assignment(
+    value: object, expected_annotations: dict[str, Any], memo: CallMemo
```

Comment:
Why are you using the object type instead of Any?

Reply:
To treat `value` as opaque. Making it an `object` makes access to any of its (non-dunder) attribute invalid. I probably should've done this in many other places too.